packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,974 @@
+  async handleOpenChange() {
+    if (this.open && !this.disabled) {
+      if (this.filteredOptions.length === 0) {
+        // Don't open the listbox if there are no options
+        this.open = false;
+        this.emit('syn-error');
+        return;
+      }
+
+      // Show
+      this.emit('syn-show');
+      this.addOpenListeners();
+
+      await stopAnimations(this);
+      this.listbox.hidden = false;
+      this.popup.active = true;
+
+      const { keyframes, options } = getAnimation(this, 'combobox.show', { dir: this.localize.dir() });
+      await animateTo(this.popup.popup, keyframes, options);
+
+      this.emit('syn-after-show');
+      return;
+    }
+
+    this.setCurrentOption(null);
+    this.displayInput.removeAttribute('aria-activedescendant');
+    // Hide
+    this.emit('syn-hide');
+    this.removeOpenListeners();
+
+    await stopAnimations(this);
+    const { keyframes, options } = getAnimation(this, 'combobox.hide', { dir: this.localize.dir() });
+    await animateTo(this.popup.popup, keyframes, options);
+    this.listbox.hidden = true;
+    this.popup.active = false;
+
+    this.emit('syn-after-hide');
+  }
+
+  /**
+   * Shows the listbox. If it is not possible to open the listbox, because there are no
+   * appropriate filtered options, a syn-error is emitted and the listbox stays closed.
+   */
+  async show() {
+    if (this.open || this.disabled) {
+      this.open = false;
+      return undefined;
+    }
+
+    this.open = true;
+    return Promise.race([waitForEvent(this, 'syn-after-show'), waitForEvent(this, 'syn-error')]);
+  }
+
+  /** Hides the listbox. */
+  async hide() {
+    if (!this.open || this.disabled) {
+      this.open = false;
+      return undefined;
+    }
+
+    this.open = false;
+    return waitForEvent(this, 'syn-after-hide');
+  }
+
+  /**
+   * Checks for validity but does not show a validation message.
+   * Returns `true` when valid and `false` when invalid.
+   */
+  checkValidity() {
+    return this.valueInput.checkValidity();
+  }
+
+  /** Gets the associated form, if one exists. */
+  getForm(): HTMLFormElement | null {
+    return this.formControlController.getForm();
+  }
+
+  /** Checks for validity and shows the browser's validation message if the control is invalid. */
+  reportValidity() {
+    return this.valueInput.reportValidity();
+  }
+
+  /** Sets a custom validation message. Pass an empty string to restore validity. */
+  setCustomValidity(message: string) {
+    this.valueInput.setCustomValidity(message);
+    this.formControlController.updateValidity();
+  }
+
+  /** Sets focus on the control. */
+  focus(options?: FocusOptions) {
+    this.displayInput.focus(options);
+  }
+
+  /** Removes focus from the control. */
+  blur() {
+    this.displayInput.blur();
+  }
+
+  private createComboboxOptionsFromQuery(queryString: string) {
+    const optgroups: SynOptGroup[] = [];
+
+    this.filteredOptions = this.getSlottedOptions()
```

Comment:
There is a smaller issue when using disabled options (which would be valid):

```html
<syn-combobox>
  <syn-option disabled>Dis</syn-option>
  <syn-option>hello</syn-option>
</syn-combobox>
```

When clicking the combobox, it will open the menu but if nothing is yet filled in, close it automatically and set the invalid state. This may happen because the first item will receive focus but can´t because its disabled. This only seems to happen when clicking **outside** of the start of the combobox (I assume outside of the input?)

Reply:
After further investigation, this seems to be a regular issue when the available size is too small: The widget will trigger the click on the option that is overlying the comboboxes input and select it. Just set your screen size really low in one of the framework demos to see this effect.

Reply:
should now be fixed for the examples. This is a problem for the combobox as well as for the select, if giving the listbox of them a fixed `max-height` via `::part`.
If there is no longer enough space for the height, we enforce it with this fix height. The solution is doing following: 

```html
max-height: min(var(--auto-size-available-height), 112px);
```
The `--auto-size-available-height` is a css variable calculated from the syn-popup, how much space for the height is left

Reply:
Seems to work fine, thank you :)